tests/ui/unit_as_impl_trait.rs

```diff
@@ -0,0 +1,45 @@
+#![warn(clippy::unit_as_impl_trait)]
+#![allow(clippy::unused_unit)]
+
```

Comment:
How about a test against the type not implementing the trait, e.g.

```rust
struct NoEq;

fn false_positive_no_trait_impl() -> impl Eq { NoEq; }
```

Reply:
Sure, will do. I'll have to investigate how to properly check that the type of the expression implements the bounds of the RPIT, probably similar to what is done in `needless_borrows_for_generic_arg` which also has to do a similar check.